votor/src/consensus_pool.rs

```diff
@@ -1025,12 +1025,42 @@ mod tests {
         assert!(pool.make_start_leader_decision(my_leader_slot, parent_slot, first_alpenglow_slot));
     }
 
-    #[test_case(Vote::new_finalization_vote(5), vec![CertificateType::Finalize(5)])]
-    #[test_case(Vote::new_notarization_vote(6, Hash::default()), vec![CertificateType::Notarize(6, Hash::default()), CertificateType::NotarizeFallback(6, Hash::default())])]
-    #[test_case(Vote::new_notarization_fallback_vote(7, Hash::default()), vec![CertificateType::NotarizeFallback(7, Hash::default())])]
-    #[test_case(Vote::new_skip_vote(8), vec![CertificateType::Skip(8)])]
-    #[test_case(Vote::new_skip_fallback_vote(9), vec![CertificateType::Skip(9)])]
-    fn test_add_vote_and_create_new_certificate_with_types(
+    #[test]
+    fn test_add_vote_and_create_new_certificate_with_types() {
+        let mut votes = vec![];
+        let mut expected_cert_types = vec![];
+
+        let slot = 5;
+        votes.push(Vote::new_finalization_vote(slot));
+        expected_cert_types.push(vec![CertificateType::Finalize(slot)]);
+
+        let slot = 6;
+        let block_id = Hash::new_unique();
+        votes.push(Vote::new_notarization_vote(slot, block_id));
+        expected_cert_types.push(vec![
+            CertificateType::Notarize(slot, block_id),
+            CertificateType::NotarizeFallback(slot, block_id),
+        ]);
+
+        let slot = 7;
+        let block_id = Hash::new_unique();
+        votes.push(Vote::new_notarization_fallback_vote(slot, block_id));
+        expected_cert_types.push(vec![CertificateType::NotarizeFallback(slot, block_id)]);
+
+        let slot = 8;
+        votes.push(Vote::new_skip_vote(slot));
+        expected_cert_types.push(vec![CertificateType::Skip(slot)]);
+
+        let slot = 9;
+        votes.push(Vote::new_skip_fallback_vote(slot));
+        expected_cert_types.push(vec![CertificateType::Skip(slot)]);
+
+        for (vote, cert_types) in votes.into_iter().zip(expected_cert_types) {
+            do_test_add_vote_and_create_new_certificate_with_types(vote, cert_types);
```

Comment:
nit: I think it's maybe actually better to have multiple do_test_add_vote_and_create_new_certificate_with_types calls:
let slot = 5;
do_test_add_vote_and_create_new_certificate_with_types(
        Vote::new_finalization_vote(slot),
        vec![CertificateType::Finalize(slot)]);
        
So when tests break it's easier to find the line which broke. But I don't feel strongly about it.